File: sdks/ruby/lib/novabilling/types/paginated_payment_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class PaginatedPaymentResponse < Internal::Types::Model
      field :data, -> { Internal::Types::Array[Novabilling::Types::PaymentResponse] }, optional: false, nullable: false
      field :meta, -> { Novabilling::Types::PaginationMeta }, optional: false, nullable: false
    end
  end
end

File: sdks/php/src/Types/PaginatedPaymentResponse.php
<?php

namespace NovaBilling\Types;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;

class PaginatedPaymentResponse extends JsonSerializableType
{
    /**
     * @var array<PaymentResponse> $data
     */
    #[JsonProperty('data'), ArrayType([PaymentResponse::class])]
    public array $data;

    /**
     * @var PaginationMeta $meta
     */
    #[JsonProperty('meta')]
    public PaginationMeta $meta;

    /**
     * @param array{
     *   data: array<PaymentResponse>,
     *   meta: PaginationMeta,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->data = $values['data'];
        $this->meta = $values['meta'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
